cmd/clusterctl/internal/test/fake_objects.go

```diff
@@ -1058,20 +1058,26 @@ func (f *FakeClusterResourceSet) Objs() []runtime.Object {
 				UID:        cluster.UID,
 			}))
 
-			binding.Spec.Bindings[cluster.Name] = addonsv1alpha3.ResourcesSetBinding{
-				Resources: map[string]addonsv1alpha3.ResourceBinding{},
+			resourceSetBinding := addonsv1alpha3.ResourceSetBinding{
+				ClusterResourceSetName: cluster.Name,
```

Comment:
Should this be `crs.Name`?

Reply:
I noticed the same CRSBinding was being used for all clusters in the test. Changed it to have 1 CRSBinding per cluster, since each cluster has its own CRSBinding that has cluster name as its name.

Reply:
Uhmm. if we are creating more binding I was expecting to see many test to fail/requiring changes.
However, here we are linking a Binding to a resource Set, and It is still strange and very confusing the fact that we are setting a field named ClusterResourceSetName = cluster.Name